lib/galaxy_test/selenium/test_histories_list.py

```diff
@@ -161,69 +141,78 @@ def test_sort_by_name(self):
+        self.components.histories.advanced_search_submit.wait_for_and_click()
+        self.assert_histories_present([])
+
+        self.components.histories.reset_input.wait_for_and_click()
+
+        # search by tags
+        self.components.histories.advanced_search_toggle.wait_for_and_click()
+        self.components.histories.advanced_search_tag_area.wait_for_and_click()
+        input_element = self.components.histories.advanced_search_tag_input.wait_for_visible()
+        input_element.send_keys(self.history3_tags[0])
+        self.send_enter(input_element)
+        self.sleep_for(self.wait_types.UX_RENDER)
+        self.components.histories.advanced_search_submit.wait_for_and_click()
+        self.assert_histories_present([self.history3_name])
 
-        self.histories_click_advanced_search()
-
-        name_filter_selector = "#input-name-filter"
-        tags_filter_selector = "#input-tags-filter"
-
-        # Search by name
-        self.set_filter(name_filter_selector, self.history2_name)
-        self.assert_grid_histories_are([self.history2_name])
-        self.unset_filter("name", self.history2_name)
-
-        self.set_filter(name_filter_selector, self.history4_name)
-        self.assert_grid_histories_are([])
-        self.unset_filter("name", self.history4_name)
-
-        # Search by tags
-        self.set_filter(tags_filter_selector, self.history3_tags[0])
-        self.assert_grid_histories_are([self.history3_name])
-        self.unset_filter("tags", self.history3_tags[0])
-
-        self.set_filter(tags_filter_selector, self.history4_tags[0])
-        self.assert_grid_histories_are([])
-        self.unset_filter("tags", self.history4_tags[0])
+    @retry_assertion_during_transitions
+    def assert_histories_present(self, expected_histories, sort_by_matters=False):
+        present_histories = self.get_present_histories()
+        assert len(present_histories) == len(expected_histories)
+        for index, row in enumerate(present_histories):
+            cell = row.find_elements(By.TAG_NAME, "td")[0]
+            if not sort_by_matters:
+                assert cell.text in expected_histories
+            else:
+                assert cell.text == expected_histories[index]
+
+    def get_present_histories(self):
+        self.sleep_for(self.wait_types.UX_RENDER)
+        return self.components.histories.histories.all()
 
     @selenium_test
     def test_tags(self):
         self._login()
         self.navigate_to_histories_page()
+        self.sleep_for(self.wait_types.UX_RENDER)
 
         # Insert a tag
-        tags_cell = self.get_history_tags_cell(self.history2_name)
-        tag_area = tags_cell.find_element(self.by.CSS_SELECTOR, ".ti-new-tag-input-wrapper input")
-        tag_area.click()
-        tag_area.send_keys(self.history2_tags[0])
-        self.send_enter(tag_area)
-
-        self.sleep_for(self.wait_types.UX_RENDER)
+        tags_cell = self.select_grid_cell("#histories-grid", self.history2_name, 3)
```

Comment:
Probably a good idea to define a variable for 3 ? Or even better have some markup that identifies what column 3 is (`data-grid-tah-column` or something like that ?)

Reply:
Moved the column to a named variable for now.